WireDomain/Sources/WireDomain/Event Processing/ConversationEventProcessor/ConversationAccessUpdateEventProcessor.swift

```diff
@@ -26,15 +27,48 @@ protocol ConversationAccessUpdateEventProcessorProtocol {
     ///
     /// - Parameter event: A conversation access update event.
 
-    func processEvent(_ event: ConversationAccessUpdateEvent) async throws
+    func processEvent(_ event: ConversationAccessUpdateEvent) async
 
 }
 
 struct ConversationAccessUpdateEventProcessor: ConversationAccessUpdateEventProcessorProtocol {
 
-    func processEvent(_: ConversationAccessUpdateEvent) async throws {
-        // TODO: [WPB-10164]
-        assertionFailure("not implemented yet")
+    let context: NSManagedObjectContext
+    let repository: any ConversationRepositoryProtocol
+
+    func processEvent(_ event: ConversationAccessUpdateEvent) async {
+        let conversationID = event.conversationID
+
+        let localConversation = await repository.fetchOrCreateConversation(
+            with: conversationID.uuid,
+            domain: conversationID.domain
+        )
+
+        let accessRoles = if let legacyAccessRole = event.legacyAccessRole {
+            getAccessRoles(from: legacyAccessRole)
+        } else {
+            event.accessRoles
+        }
+
+        await context.perform {
+            localConversation.accessModeStrings = event.accessModes.map(\.rawValue)
+            localConversation.accessRoleStringsV2 = accessRoles.map(\.rawValue)
```

Comment:
Shouldn't this be part of the repository?

Reply:
added a local store dedicated method for that specific object update in https://github.com/wireapp/wire-ios/pull/2028/commits/289eab5fa5f60942c24c2962717ac97c029e8a5c